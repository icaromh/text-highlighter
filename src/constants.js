export const COLOR_1 = '#ff8a80'
export const COLOR_2 = '#fff176'
export const COLOR_3 = '#4db6ac'
