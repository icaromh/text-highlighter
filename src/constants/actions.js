export const ADD_HIGHLIGHT = 'ADD_HIGHLIGHT'
export const COPY_DOCUMENT = 'COPY_DOCUMENT'
export const FILTER_HIGHLIGHTS = 'FILTER_HIGHLIGHTS'
export const TOGGLE_FILTER_HIGHLIGHT = 'TOGGLE_FILTER_HIGHLIGHT'
