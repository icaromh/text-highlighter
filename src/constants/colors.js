export const RED = '#ff8a80'
export const YELLOW = '#fff176'
export const GREEN = '#4db6ac'
