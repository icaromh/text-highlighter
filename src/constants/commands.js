export const HIGHLIGHT_CMD = 'hiliteColor'
export const COPY_CMD = 'copy'
export const SELECT_ALL_CMD = 'selectAll'
