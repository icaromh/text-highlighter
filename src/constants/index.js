const TWITTER_URL_SHARE = sentence => (
  `https://twitter.com/intent/tweet?text=${sentence}&tw_p=tweetbutton&via=icaromh`
)

export default TWITTER_URL_SHARE
