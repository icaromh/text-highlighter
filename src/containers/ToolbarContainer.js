import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import { CopyIcon } from '../components/Icons'
import Button from '../components/Button'
import MarkButton from '../components/Button/MarkButton'
import Toolbar from '../components/Toolbar'

import {
  RED,
  YELLOW,
  GREEN,
} from '../constants/colors'

import {
  addHighlight,
  copyDocument,
} from '../actions'


const ToolbarContainer = ({ handleAddHighlight, handleCopyContent }) => (
  <Toolbar>
    <MarkButton onClick={() => handleAddHighlight(RED)} color={RED} />
    <MarkButton onClick={() => handleAddHighlight(YELLOW)} color={YELLOW} />
    <MarkButton onClick={() => handleAddHighlight(GREEN)} color={GREEN} />

    <Button title="Copy the document as showed" onClick={() => handleCopyContent()}>
      <CopyIcon />
    </Button>
  </Toolbar>
)

ToolbarContainer.propTypes = {
  handleAddHighlight: PropTypes.func.isRequired,
  handleCopyContent: PropTypes.func.isRequired,
}

export {
  ToolbarContainer,
}

export default connect(
  null,
  {
    handleAddHighlight: addHighlight,
    handleCopyContent: copyDocument,
  },
)(ToolbarContainer)
