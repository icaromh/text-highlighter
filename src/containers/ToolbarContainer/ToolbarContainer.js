import React from 'react'
import { connect } from 'react-redux'
import { addHighlight } from '../../actions'

import { CopyIcon } from '../../components/Icons'
import Button, { MarkButton } from '../../components/Button'
import { RED, YELLOW, GREEN } from '../../constants/colors'

const ToolbarContainer = ({ addHighlight }) => (
  <nav className="Toolbar">
    <MarkButton onClick={() => addHighlight(RED)} color={RED} />
    <MarkButton onClick={() => addHighlight(YELLOW)} color={YELLOW} />
    <MarkButton onClick={() => addHighlight(GREEN)} color={GREEN} />

    <Button title="Copy the document as showed" onClick={() => console.log('copied')}>
      <CopyIcon />
    </Button>
  </nav>
)

export default connect(
  null,
  { addHighlight }
)(ToolbarContainer)
