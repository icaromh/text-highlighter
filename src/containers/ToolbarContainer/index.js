import ToolbarContainer from './ToolbarContainer'
import './Toolbar.css'

export default ToolbarContainer
