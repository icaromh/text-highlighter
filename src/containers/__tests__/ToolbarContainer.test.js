import React from 'react'
import { shallow } from 'enzyme'

import { ToolbarContainer } from '../ToolbarContainer'
import { RED, YELLOW, GREEN } from '../../constants/colors'

describe('#ToolbarContainer', () => {
  test('expect to render', () => {
    const wrapper = shallow(
      <ToolbarContainer
        handleAddHighlight={jest.fn()}
        handleCopyContent={jest.fn()}
      />,
    )

    expect(wrapper).toMatchSnapshot()
  })

  test('should call handleAddHighlight on mark click', () => {
    const handleAddHighlight = jest.fn()
    const wrapper = shallow(
      <ToolbarContainer
        handleAddHighlight={handleAddHighlight}
        handleCopyContent={jest.fn()}
      />,
    )

    wrapper.find('MarkButton').at(0).simulate('click')
    wrapper.find('MarkButton').at(1).simulate('click')
    wrapper.find('MarkButton').at(2).simulate('click')

    expect(handleAddHighlight).toBeCalled()
    expect(handleAddHighlight.mock.calls[0][0]).toBe(RED)
    expect(handleAddHighlight.mock.calls[1][0]).toBe(YELLOW)
    expect(handleAddHighlight.mock.calls[2][0]).toBe(GREEN)
  })

  test('should call handleAddHighlight on mark click', () => {
    const handleCopyContent = jest.fn()
    const wrapper = shallow(
      <ToolbarContainer
        handleAddHighlight={jest.fn()}
        handleCopyContent={handleCopyContent}
      />,
    )

    wrapper.find('Button').at(0).simulate('click')
    expect(handleCopyContent).toBeCalled()
  })
})
