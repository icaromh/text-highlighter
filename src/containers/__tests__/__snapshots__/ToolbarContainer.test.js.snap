// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#ToolbarContainer expect to render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ToolbarContainer
    handleAddHighlight={[MockFunction]}
    handleCopyContent={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <MarkButton
          color="#ff8a80"
          onClick={[Function]}
        />,
        <MarkButton
          color="#fff176"
          onClick={[Function]}
        />,
        <MarkButton
          color="#4db6ac"
          onClick={[Function]}
        />,
        <Button
          onClick={[Function]}
          title="Copy the document as showed"
        >
          <CopyIcon />
        </Button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "color": "#ff8a80",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "color": "#fff176",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "color": "#4db6ac",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <CopyIcon />,
          "onClick": [Function],
          "title": "Copy the document as showed",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <MarkButton
            color="#ff8a80"
            onClick={[Function]}
          />,
          <MarkButton
            color="#fff176"
            onClick={[Function]}
          />,
          <MarkButton
            color="#4db6ac"
            onClick={[Function]}
          />,
          <Button
            onClick={[Function]}
            title="Copy the document as showed"
          >
            <CopyIcon />
          </Button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "color": "#ff8a80",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "color": "#fff176",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "color": "#4db6ac",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <CopyIcon />,
            "onClick": [Function],
            "title": "Copy the document as showed",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
